src/docs.rs

```diff
@@ -0,0 +1,567 @@
+// autogenerated. Please look at /tests/check-docs.rs
+
+#[allow(clippy::too_many_lines)]
+#[rustfmt::skip]
+fn explain(lint: &str) {
+    println!("{}", match lint {
+        "absurd_extreme_comparisons" => "### What it does\nChecks for comparisons where one side of the relation is\neither the minimum or maximum value for its type and warns if it involves a\ncase that is always true or always false. Only integer and boolean types are\nchecked.\n### Why is this bad?\nAn expression like `min <= x` may misleadingly imply\nthat it is possible for `x` to be less than the minimum. Expressions like\n`max < x` are probably mistakes.\n### Known problems\nFor `usize` the size of the current compile target will\nbe assumed (e.g., 64 bits on 64 bit systems). This means code that uses such\na comparison to detect target pointer width will trigger this lint. One can\nuse `mem::sizeof` and compare its value or conditional compilation\nattributes\nlike `#[cfg(target_pointer_width = \"64\")] ..` instead.\n### Example\n\nlet vec: Vec<isize> = Vec::new();\nif vec.len() <= 0 {}\nif 100 > i32::MAX {}\n",
```

Comment:
Having this file in the repository will probably result in many merge conflicts in PRs that touch documentation in one way or another.

Reply:
Ok, I'm going to solve this by creating a lot of `src/docs/*.md`  files and using `include_str!()` on each. That way we only get conflicts if two lints are added/removed (and those are easy to fix) or two people work on the same lint docs (but we get that already).